pwnagotchi/plugins/default/clock.py

```diff
@@ -0,0 +1,147 @@
+    __license__ = "GPL3"
+    __plugin__ = "clock"
+    __description__ = "Shows the time, sometimes even the current one."
+
+    def __init__(self):
+        self.synced = False
+
+    def timedatectl_state(self):
+        try:
+            state = subprocess.run(
+                ["timedatectl", "status"],
+                capture_output=True, check=True, text=True, timeout=5
+            )
+
+            matches = re.search(r".*System clock synchronized: (\w*)\W", state.stdout, re.MULTILINE)
+            if matches and matches.group(1) == "yes":
+                logging.info(f"[{self.__plugin__}]: system clock is in sync")
+                self.synced = True
+
+            return self.synced
+
+        except TimeoutExpired as error:
+            logging.error(f"[{self.__plugin__}] timeout: {error}")
+        except CalledProcessError as error:
+            logging.error(f"[{self.__plugin__}] error: {error}")
+
+    def on_loaded(self):
+        logging.info(f"[{self.__plugin__}]: plugin loaded")
+
+    def on_ready(self, agent):
+        self.timedatectl_state()
+        # try:
+        #     state = subprocess.run(
+        #         ["timedatectl", "status"],
+        #         capture_output=True, check=True, text=True, timeout=5
+        #     )
+
+        #     matches = re.search(r".*System clock synchronized: (\w*)\W", state.stdout, re.MULTILINE)
+        #     if matches and matches.group(1) == "yes":
+        #         logging.info(f"[{self.__plugin__}]: system clock is synchronized")
+        #         self.synced = True
+
+        # except TimeoutError as error:
+        #     logging.error(f"[{self.__plugin__}] timeout: {error}")
+        # except CalledProcessError as error:
+        #     logging.error(f"[{self.__plugin__}] error: {error}")
+
+    def on_ui_setup(self, ui):
+        if ui.is_waveshare_v2():
+            pos = (208, 67)
+        elif ui.is_inky():
+            # guessed values, add tested ones if you can
+            pos = (112, 30)
+        else:
+            # guessed values, add tested ones if you can
+            pos = (127, 51)
+
+        label_spacing = 0
+
+        ui.add_element(
+            "clock",
+            LabeledValue(
+                color=BLACK,
+                label="",
+                value="-:-:-",
+                position=pos,
+                label_font=fonts.Small,
+                text_font=fonts.Small,
+                label_spacing=label_spacing,
+            ),
+        )
+
+        if self.synced:
+            ui.set("clock", f"{datetime.now().time().isoformat(timespec='seconds')} ")
+        elif self.query_ntp():
+            ui.set("clock", f"{self.query_ntp()} ")
+
+    def on_ui_update(self, ui):
+        if self.synced:
+            ui.set("clock", f"{datetime.now().time().isoformat(timespec='seconds')} ")
+        else:
+            ui.set("clock", f"-:-:- ")
+
+    def on_wait(self, agent, t):
+        if not self.synced:
+            self.query_ntp()
```

Comment:
I don't see the point to these extra calls. It seems like query_ntp() just returns the time according to NTP and doesn't set any actual state. If you're just throwing away the result why do it at all?

Reply:
Same goes for the other state handlers, like bored.